File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
  public int maxSubArray(int[] nums) {
    int prevMax = -10001;
    int currMax = -10001;
    int ans = -10001;
    for (int n : nums) {
      prevMax = currMax;
      currMax = Math.max(n, n + prevMax);
      ans = Math.max(ans, currMax);
    }
    return ans;
  }
}

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
  let [prevMax, currMax, ans] = [-Infinity, -Infinity, -Infinity];
  
  nums.forEach(n => {
    prevMax = currMax;
    currMax = Math.max(n, n + prevMax);
    ans = Math.max(ans, currMax);
  });
  
  return ans;
};